tests/unit/service/test_resourcedriver.py

```diff
@@ -23,15 +23,15 @@ def test_merge(self):
         result = merger.merge(
             PropValueMap({
                 'propA': {'type': 'string', 'value': 'propA'}, 
-                'propB': {'type': 'string', 'value': 'propB'}
+                'propB': {'type': 'integer', 'value': 1}
             }),
             PropValueMap({
                 'propA': {'type': 'string', 'value': 'sysPropA'}
             })
         )
         self.assertEqual(result, PropValueMap({
             'propA': {'type': 'string', 'value': 'propA'}, 
-            'propB': {'type': 'string', 'value': 'propB'},
+            'propB': {'type': 'integer', 'value': 1},
             'system_propA': {'type': 'string', 'value': 'sysPropA'}
```

Comment:
Is it worth checking that at least one of the non-primitive types e.g. map, list works, or is that overkill?

Reply:
Good point. The type should be fairly unimportant to the merge but I will put it in anyway because it'll catch any regression.